src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1823,6 +1831,26 @@ FixedwingPositionControl::tecs_update_pitch_throttle(float alt_sp, float airspee
 					     accel_body, (_global_pos.timestamp > 0), in_air_alt_control,
 					     _global_pos.alt, _local_pos.v_z_valid, _local_pos.vz, _local_pos.az);
 
+	/* scale throttle cruise by baro pressure */
+	if (_parameters.throttle_alt_scale > FLT_EPSILON) {
+
+		bool baro_updated = false;
+		orb_check(_sensor_baro_sub, &baro_updated);
+
+		sensor_baro_s baro;
+
+		if (orb_copy(ORB_ID(sensor_baro), _sensor_baro_sub, &baro) == PX4_OK) {
+			if (PX4_ISFINITE(baro.pressure) && PX4_ISFINITE(_parameters.throttle_alt_scale)) {
+				// scale throttle as a function of sqrt(p0/p) (~ EAS -> TAS at low speeds and altitudes ignoring temperature)
+				const float eas2tas = sqrtf(MSL_PRESSURE_MILLIBAR / baro.pressure);
+				const float scale = constrain(eas2tas * _parameters.throttle_alt_scale, 0.9f, 1.8f);
+
+				throttle_cruise = throttle_cruise * scale;
+				throttle_max = throttle_max * scale;
```

Comment:
limit throttle_max upper value to 1.0 and limit upper value of throttle_cruise to be lower than throttle_max.